rawfile/orchestrator/k8s.py

```diff
@@ -229,3 +127,62 @@ def write_config_map(name: str, key: str, value: str, overwrite=False):
             logger.trace("Created configmap with key", name=name, key=key, value=value)
         else:
             raise e
+
+
+class NodeIPMapping:
+    def __init__(self):
+        if not config.csi_driver:
+            raise Exception("Should run from CSI Driver")
+        self.apps_v1 = k8s_client.AppsV1Api()
+        self.core_v1 = k8s_client.CoreV1Api()
+        ds = self.apps_v1.read_namespaced_daemon_set(
+            namespace=config.namespace, name=config.csi_driver.node_ds
+        )
+        self.selector = ds.spec.selector.match_labels
+        self.label_selector = ",".join(f"{k}={v}" for k, v in self.selector.items())
+
+        self._mapping = {}
+        self.reload_mapping()
+
+        # Start watcher in background
+        self._stop_event = threading.Event()
+        self._thread = threading.Thread(target=self.watch_daemonset_pods, daemon=True)
+        self._thread.start()
+
+    def reload_mapping(self):
+        pods = self.core_v1.list_namespaced_pod(
+            namespace=config.namespace, label_selector=self.label_selector
+        )
+        mapping = {}
+        for pod in pods.items:
+            if pod.status.phase == "Running" and pod.status.pod_ip:
+                mapping[pod.spec.node_name] = pod.status.pod_ip
+        self._mapping = mapping
+
+    @property
+    def mapping(self):
+        return self._mapping
+
+    def watch_daemonset_pods(self):
+        w = watch.Watch()
+        while not self._stop_event.is_set():
+            try:
+                for _ in w.stream(
+                    self.core_v1.list_namespaced_pod,
```

Comment:
could we somehow leverage the csinode CR instead?

Reply:
I have used the pod selector of the daemonset since I need the IP address of the pod itself

You mean that we can watch csinode resources and get ip of the pod related to that node? It will make it complicated I think